zimui/src/components/TopicSection.vue

```diff
@@ -0,0 +1,137 @@
+<script setup lang="ts">
+import { ref, getCurrentInstance } from 'vue'
+import TopicCard from '../components/TopicCard.vue'
+
+defineProps({
+  data: {
+    type: Object,
+    required: true,
+  },
+})
+const instance = getCurrentInstance()
+const uid = ref('carroussel_' + instance?.uid)
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const splitChunks = (inputArray: any[], perChunk: number) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return inputArray.reduce((all: any[], one: ConcatArray<never>, i: number) => {
+    const ch = Math.floor(i / perChunk)
+    all[ch] = [].concat(all[ch] || [], one)
+    return all
+  }, [])
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const limitCardsPerSections = (inputArray: any[], section_slug: string) => {
+  const maxCardPerSection = 10
+  if (inputArray.length > maxCardPerSection) {
+    const inputSliced = inputArray.slice(0, maxCardPerSection)
+    inputSliced.push({
+      kind: 'more',
+      count_more: inputArray.length - maxCardPerSection,
+      slug: section_slug,
+    })
+    return inputSliced
+  }
+  return inputArray
+}
+
+const getSelector = (value: string) => {
+  return '#' + value
+}
+</script>
+
+<template>
+  <div class="subsection pt-3 pb-3">
+    <div class="container pt-3">
+      <router-link
+        class="text-decoration-none text-reset"
+        :to="`./${data.slug}`"
+      >
+        <h4 :class="{ 'mb-2': data.description, 'mb-4': !data.description }">
+          <span>
+            {{ data.title }}
+            <span class="right-arrow">
+              <FontAwesomeIcon
+                aria-label="Arrow Right icon"
+                icon="fa-solid fa-arrow-right"
+              />
+            </span>
+          </span>
+        </h4>
+        <div v-if="data.description" class="lead mb-2">
+          <span class="description" :title="data.description">
+            {{ data.description }}
+          </span>
+        </div>
+      </router-link>
+    </div>
+    <div :id="uid" class="carousel slide">
+      <button
+        class="carousel-control-prev"
+        type="button"
+        :data-bs-target="getSelector(uid)"
+        data-bs-slide="prev"
+      >
+        <span class="carousel-control-prev-icon" aria-hidden="true"></span>
+        <span class="visually-hidden">Previous</span>
+      </button>
+      <div class="carousel-inner">
+        <div
+          v-for="(chunk, chunkIndex) in splitChunks(
+            limitCardsPerSections(data.subsections, data.slug),
+            $grid.lg ? 4 : $grid.sm ? 2 : 1,
```

Comment:
How's this computed? Dynamically? Where is it defined?

Reply:
This is the magic of [vue-screen](https://github.com/reegodev/vue-screen)